src/m365/planner/commands/plan/plan-add.ts

```diff
@@ -0,0 +1,167 @@
+import { Logger } from '../../../../cli';
+import {
+  CommandOption
+} from '../../../../Command';
+import GlobalOptions from '../../../../GlobalOptions';
+import request from '../../../../request';
+import Utils from '../../../../Utils';
+import GraphCommand from '../../../base/GraphCommand';
+import commands from '../../commands';
+
+interface CommandArgs {
+  options: Options;
+}
+
+interface Options extends GlobalOptions {
+  title: string;
+  ownerGroupId?: string;
+  ownerGroupName?: string;
+}
+
+class GraphPlannerPlanAddCommand extends GraphCommand {
+  public get name(): string {
+    return commands.PLANNER_PLAN_ADD;
+  }
+
+  public get description(): string {
+    return 'Adds a new Microsoft Planner plan';
+  }
+
+  public getTelemetryProperties(args: CommandArgs): any {
+    const telemetryProps: any = super.getTelemetryProperties(args);
+    telemetryProps.ownerGroupId = typeof args.options.ownerGroupId !== 'undefined';
+    telemetryProps.ownerGroupName = typeof args.options.ownerGroupName !== 'undefined';
+    return telemetryProps;
+  }
+
+  public defaultProperties(): string[] | undefined {
+    return ['id', 'title', 'createdDateTime', 'owner'];
+  }
+
+  public commandAction(logger: Logger, args: CommandArgs, cb: () => void): void {
+    const endpointPlanner: string = `${this.resource}/v1.0/planner/plans`;
+    if (args.options.ownerGroupId) {
+      this.getGroupsWithFilter(`?$filter=ID eq '${args.options.ownerGroupId}'`)
```

Comment:
Since we already have a group ID, is there added value in checking if it exists? The command will fail either way and the returned error should make it clear what's missing. By removing this code, we can significantly speed up the command and simplify its code.